### DISCUSSION TITLE
IntegrateBC in Example 27 with generic alpha

### DISCUSSION THREAD

**Discussion by faircdl** [mfem/mfem#2205] [Q&A]
I'm new to MFEM (a two-day veteran!) and have been working through the examples and extending them to problems I'm interested in solving.  I've partly extended example 27 for cylindrical coordinates, but I'm wondering how to extend IntegrateBC to use the same FunctionCoefficient I defined for the bilinear integrator to properly perform the integral over the boundary.  

Specifically, I'm solving the PDE: -div(a * x * grad(u)) = x * g, where a is a constant, x is the spatial cartesian coordinate, u is the unknown, and g is some function.  If I map to a standard form of the PDE: -div(alpha * grad(u)) = f then I have alpha = a*x and f = x * g.  I have successfully using the FunctionCoefficient type to handle alpha and f for the bilinear and linear form integrators.  However, IntegrateBC is a little above my MFEM skill level right now.  I'm familiar with FEM and sorta see what's going on, but I'm not exactly sure how to incorporate alpha as a FunctionCoefficient instead of just a double.  

If I'm missing a similar discussion on another thread or an example, please point me to it.  Otherwise, thanks in advance for the help.  MFEM is an incredible package.

**Comment by vladotomov** [comment]
If `alpha` is a `FunctionCoefficient`, you can simply call `alpha.Eval(...)` at the current integration point.

Sorry for the late reply. Posting questions in the `Issues` section has higher probability of getting attention.

Best,
Vladimir
